lib/application_insights/telemetry_client.rb

```diff
@@ -217,7 +223,22 @@ def track_request(id, start_time, duration, response_code, success, options={})
         :measurements => options[:measurements] || {}
       )
 
-      self.channel.write(data, self.context)
+      self.channel.write(data, self.context, start_time)
+    end
+
+    # Sends a single telemetry.
+    # This is useful in cases like telemetry forwarding, where the data is already in the
+    # format of envelope and only want to leverage the telemetry channel to send data out.
+    # @param [Hash] data the envelope data of the telemetry.
+    def track_envelope(data)
```

Comment:
I actually have some concerns about this API. It's very different from other SDKs, where the API is trackTelemetry() but not trackEnvelope(). I added that originally so I can easily take advantage of this library in my scenario, i.e., log forwarding. But now I don't feel good about it.
I'd like to revert it (and also the ```options[:time]```). So the trackRequest/Dependency... API is called for a new event.
For log forwarding scenario, it's done by constructing the Telemetry object, setting the corresponding context, timestamp, and call trackTelemetry().